pkg/controller/sidecarset/sidecarset_processor.go

```diff
@@ -191,6 +195,23 @@ func (p *Processor) updatePods(control sidecarcontrol.SidecarControl, pods []*co
 	return nil
 }
 
+func (p *Processor) updateSidecarSetUpgradeState(sidecarSet *appsv1alpha1.SidecarSet, pods []*corev1.Pod,
+	sideCarSetUpgradeState sidecarcontrol.SidecarSetUpgradeState) error {
+	for _, pod := range pods {
+		podClone := pod.DeepCopy()
+		err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			sidecarcontrol.UpdatePodSidecarSetHashState(podClone, sidecarSet, sideCarSetUpgradeState)
+			return p.Client.Update(context.TODO(), podClone)
```

Comment:
Please use patch pod, you can ref code: https://github.com/openkruise/rollouts/blob/master/pkg/trafficrouting/network/ingress/ingress.go#L136